src/Resources/Resources/ActiveDirectory/Cmdlets/NewAzureADServicePrincipalCommand.cs

```diff
@@ -342,7 +341,14 @@ private void CreateSimpleServicePrincipal()
             if (ShouldProcess(target: createParameters.ApplicationId.ToString(), action: shouldProcessMessage))
             {
                 PSADServicePrincipalWrapper servicePrincipal = new PSADServicePrincipalWrapper(ActiveDirectoryClient.CreateServicePrincipal(createParameters));
-                servicePrincipal.Secret = Password;
+                if(printPassword)
+                {
+                    servicePrincipal.Secret = Password;
+                }
+                else if(printUseExistingSecret)
+                {
+                    WriteVerbose(String.Format("No new secret was created. This ServicePrincipal will use the password associated with application {0}.", ApplicationId));
```

Comment:
So, I know they don't follow this elsewhere in the code, but the best practice is to put any significnat string messages that are displayed to customers into the resources file for this assembly, rather than including them in the code.  This makes translation of the product possible in the future.

Reply:
ALso, please add a change to the changelog

Reply:
All done.